src/common/task.h

```diff
@@ -13,6 +13,73 @@
 
 typedef uint8_t TaskSpec;
 
+class TaskExecutionSpec {
+ public:
+  TaskExecutionSpec(const std::vector<ObjectID> &execution_dependencies,
+                    TaskSpec *spec,
+                    int64_t task_spec_size);
+  TaskExecutionSpec(TaskExecutionSpec *execution_spec);
+
+  /// Get the task's execution dependencies.
+  ///
+  /// @return A vector of object IDs representing this task's execution
+  ///         dependencies.
+  std::vector<ObjectID> ExecutionDependencies();
```

Comment:
You've obviously given this more thought, but it seems cleaner to me to expose two methods `ObjectIDDependencies` and `ExecutionDependencies`.

There are some places in the local scheduler where we would have to iterate over both lists (e.g., when issuing reconstruction commands).

However, there are some places where we only want to iterate over a single list, such as when issuing fetch requests).

@pcmoritz thoughts about this?

Reply:
E.g., in the future it could make sense to change the dummy object ID implementation and have execution dependencies actually be a vector of `TaskID`s instead of `ObjectID`s.

Reply:
Hmm, I'm not convinced that it would be cleaner to have a vector of `TaskID`s instead of `ObjectID`s. It would be nice to keep reconstruction and fetching on relatively the same path. If you're worried about the latency of having to go to the result table, I think it makes more sense to cache information for actor tasks at the local scheduler.

Reply:
I agree it would be cleaner to separate ObjectIDDependencies from ExecutionDependencies. Probably also having TaskIDs instead of ObjectIDs for the latter but we can think more about that and do it in a followup PR.